src/main/java/duke/Parser.java

```diff
@@ -34,75 +34,74 @@ public Command parse(String line) throws CommandParseException {
         } else {
             String c = line.substring(0, argWhitespace).toLowerCase();
             switch (c) {
-                case "mark":
+            case "mark":
+                try {
+                    int taskIndex = Integer.parseInt(line.substring(argWhitespace + 1)) - 1;
+                    return new MarkCommand(taskIndex);
+                } catch (NumberFormatException e) {
+                    throw new CommandParseException("You did not provide a number!", line);
+                }
+            case "unmark":
+                try {
+                    int taskIndex = Integer.parseInt(line.substring(argWhitespace + 1)) - 1;
+                    return new UnmarkCommand(taskIndex);
+                } catch (NumberFormatException e) {
+                    throw new CommandParseException("You did not provide a number!", line);
+                }
+            case "delete":
+                try {
+                    int taskIndex = Integer.parseInt(line.substring(argWhitespace + 1)) - 1;
+                    return new DeleteCommand(taskIndex);
+                } catch (NumberFormatException e) {
+                    throw new CommandParseException("You did not provide a number!", line);
+                }
```

Comment:
I like how you are following the coding standard for the switch-case statements 😄 .